packages/web/src/js/utils/ComponentFunctions.ts

```diff
@@ -8,34 +8,47 @@ type DataTriggerAttribute = 'data-toggle' | 'data-dismiss';
 const ATTRIBUTE_DATA_TOGGLE = `data-toggle`;
 const ATTRIBUTE_DATA_DISMISS = `data-dismiss`;
 
+const onClickHandler = (element: HTMLElement, component: typeof BaseComponent, method: string, event: Event) => {
```

Comment:
Please include in this implementation also this PR - https://github.com/lmc-eu/spirit-design-system/pull/518

The reason for this is to pass the main component element into the class. So in `this.element` will be the component itself, eg. `Collapse` and not its trigger element. It should be also simpler to work with it in the class itself (no need for searching for data attributes.)